### PR TITLE
add reusable watcher saga

### PR DESCRIPTION
**Pull request by meho99**

### CODE REVIEW

File: src/store/utils.ts [hunk 1]
```diff
@@ -25,3 +30,44 @@ export const convertForMulticall = <
 
   return contractCall;
 };
+
+type CreateWatcherSagaOptions = {
```
Comment (mwiniarsk) [inline src/store/utils.ts:34]: shouldn't utils have their own types?

Comment (michalsidzej) [reply src/store/utils.ts:34]: Or maybe we can put it in `src/store/test.ts`

**Review by michalsidzej — DISMISSED** [review]
otherwise, LGTM

File: src/store/utils.ts (hunk 1 above)
Comment (meho99) [reply src/store/utils.ts:34]: > Or maybe we can put it in `src/store/test.ts`

You mean `src/store/types.ts`, right? If so, I totally agree

**Pull request merged**